open_humans/account_views.py

```diff
@@ -141,3 +174,149 @@ def delete(self, request, *args, **kwargs):
+        if not token:
+            return redirect('account-password-reset-fail')
+        return super().dispatch(request, uidb36, key, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        ret = super().get_context_data(**kwargs)
+        ret['action_url'] = reverse(
+            'account_reset_password_from_key',
+            kwargs={'uidb36': self.kwargs['uidb36'],
+                    'key': self.kwargs['key']})
+        return ret
+
+    def change_password(self, password):
+        user = self.reset_user
+        user.set_password(password)
+        user.save()
+        return user
+
+    def form_valid(self, form):
+        if form.is_valid():
+            self.change_password(form.clean_password())
+
+            perform_login(
+                self.request, self.reset_user,
+                email_verification=EMAIL_VERIFICATION)
+            member = Member.objects.get(user=self.reset_user)
+            next_url = member.password_reset_redirect
+            member.password_reset_redirect = '' # Clear redirect from db
+            member.save()
+            messages = {
+                'settings_updated': {
+                    'level': django_messages.SUCCESS,
+                    'text': 'Password successfully reset.'},}
+            return redirect(unquote_plus(next_url))
+
+
+class PasswordChangeView(PasswordChangeView):
+    """
+    Change the password
+    """
+    template_name = 'account/password_change.html'
+    form_class = ChangePasswordForm
+    success_url = reverse_lazy('my-member-settings')
+    def form_valid(self, form):
+
+        return super().form_valid(form)
+
+
+class ConfirmEmailView(ConfirmEmailView):
+    """
+    Subclass ConfirmEmailView to set the user email.
+    """
+
+    def post(self, *args, **kwargs):
+        """
+        Replace allauth's ConfirmEmailView.post because we want to do a few more
+        things.
+        """
+        self.object = confirmation = self.get_object()
+        confirm = confirmation.confirm(self.request)
+        try:
+            # For now, deleting additional email addresses as we only support one
+            # This has the advantage that if a user wants to change back, they
+            # can do so, and also reduces database clutter
+            queryset = EmailAddress.objects.filter(user=confirm.user)
+            queryset.exclude(email=confirm.email).all().delete()
+
+            # Set new email to primary
+            new_email = queryset.get(email=confirm.email)
+            new_email.primary = True
+            new_email.save()
+        except AttributeError:
+            pass # If someone tries to use an expired or incorrect key,
+                 # let allauth's error handling handle it
+
+        django_messages.add_message(
+            self.request,
+            django_messages.SUCCESS,
+            'account/messages/email_confirmed.txt',
+            {'email': confirmation.email_address.email})
+        resp = self.login_on_confirm(confirmation)
```

Comment:
it seems like this could instead be configured with `ACCOUNT_LOGIN_ON_EMAIL_CONFIRMATION`?

Reply:
Deleted